UIKitVK/UIKitVK/Controller/NewsViewController.swift

```diff
@@ -21,30 +22,59 @@ final class NewsViewController: UIViewController {
 
     private let networkService = NetworkService()
     private var news: [NewsItem] = []
+    private var refreshControl = UIRefreshControl()
+    private var isLoading = false
+    private var nextPage = String()
+    private var currentInterval = 0
     private lazy var photoCacheService = PhotoCacheService(container: newsTableView)
 
     // MARK: - Life Cycle
 
     override func viewDidLoad() {
         super.viewDidLoad()
-        fetchNews()
+        setupUI()
     }
 
     // MARK: - Private Methods
 
+    private func setupUI() {
+        createRefreshControl()
+        newsTableView.prefetchDataSource = self
+    }
+
+    private func createRefreshControl() {
+        guard let colorTitle = UIColor(named: Constants.CustomColorNames.linkCustomColorName) else { return }
+        let refreshControlAttributes: [NSAttributedString.Key: UIColor] =
+            [.foregroundColor: colorTitle]
+        refreshControl.attributedTitle = NSAttributedString(
+            string: Constants.OtherConstants.refreshControlTitle,
+            attributes: refreshControlAttributes
+        )
+        refreshControl.tintColor = colorTitle
+        refreshControl.addTarget(self, action: #selector(refreshNewsAction), for: .valueChanged)
+        newsTableView.addSubview(refreshControl)
+    }
+
     private func fetchNews() {
-        networkService.fetchPostNews { [weak self] news in
+        var mostFreshDate: TimeInterval?
+        if let firstNews = news.first {
+            mostFreshDate = Double(firstNews.date) + 1
+        }
+        networkService.fetchPostNews(startTime: mostFreshDate ?? 0) { [weak self] news in
             guard let self = self else { return }
             switch news {
             case let .success(data):
-                self.updateNewsTable(response: data)
+                self.news = self.updateNewsTable(response: data) + self.news
+                self.currentInterval = data.news.first?.date ?? 0
+                self.nextPage = data.nextPage ?? Constants.OtherConstants.emptyString
```

Comment:
```suggestion
                self.currentInterval = data.news.first?.date ?? 0
                self.nextPage = data.nextPage ?? Constants.OtherConstants.emptyString
```
можно в guard обернуть